Fix plt motionfields

dnerini opened the pull request:
Addresses issue #165, namely

* update interface to plot_geography()
* update interface for quiver() and streamplot() functions for consistency with plot_precip_fields()
* xpixelsize and ypixelsize are not required anymore
* update docstrings
* add tests

dnerini committed:
Fix interface to plot_geography

dnerini committed:
Fix interface to plot_geography (part 2)

dnerini committed:
Fix docstrings

dnerini committed:
Fix plotting functions for motion fields

dnerini committed:
Remove pixelsize as a required argument

aperezhortal commented on pysteps/visualization/motionfields.py:
```diff
@@ -238,11 +209,9 @@ def streamplot(
     drawlonlatlines=False,
-    basemap_resolution="l",
-    cartopy_scale="50m",
     lw=0.5,
```
**lw** keyword is not used in streamplot. Perhaps it should be added to the plot_geography call in streamplot? (like in quiver).

dnerini replied:
well spotted! corrected in aaee5d1

aperezhortal commented on pysteps/visualization/motionfields.py:
```diff
@@ -26,11 +25,10 @@ def quiver(
     map=None,
```
Using **map** directly as keyword masked the build-in `map` python function. Although the code works (since the reference to map is redefined) maybe we can move map to kwargs and assign its value to a different variable name `plot_map=kwargs.pop("map")`. By doing this we don't need to change the interface and we avoid the shadowing of the python `map` function. What do you think?

dnerini replied:
It's an excellent point, but unfortunately this is true not only for the quiver() function, but also for all our plotting methods.

We could implement your workaround for most cases,  but I feel that this would somewhat complicate the code. 
Alternatively we could explicitly rename the `map` argument to `plot_map`, but this will change the interface.

I understand that the current situation is not ideal, but do you see any major problem in leaving it as it is? Our local `map` argument shadows the built-in `map` method, which is not used in there. Do you think that something can go wrong by doing it?

aperezhortal replied:
> Do you think that something can go wrong by doing it?

I don't think so. We can leave it as it is. But in the long term, maybe we can change the interface slowly, first adding a warning that the argument name will change, and a few releases down the road, change the implementation. 
Either way, this is not an important issue to solve right now, but we should keep it in mind for future releases :)

dnerini replied:
Right, good idea. In cc505e2  I introduced a FutureWarning and internally renamed `map` to `plot_map`, do you mind having a quick look to check whether this is OK?

aperezhortal replied:
Great! I also added some minor changes to the `plot_geography` interface in commits 9714c3f and 9d0ed8a.

aperezhortal commented:
Great job! I added two minor comments in the file changes. 

Also, I think that the following statement In motionfields.py, lines 332-333, have no effect, since X,Y are redefined later on. Could be? 
```
if not reproject:
    X, Y = np.meshgrid(x, y)
```

dnerini committed:
Add missing 'lw' argument

dnerini committed:
Remove unnecessary lines

dnerini commented:
Thanks for the review! 

(indeed, lines 332-333 were unnecessary, see 0ec44ee )

dnerini committed:
Add FutureWarning and internally rename 'map' to 'plot_map'

aperezhortal committed:
Rename map argument

- Since map was an argument and not a keyword, it can be renamed safely.
- The keywords arguments were explicitly declared in the plot_geography calls.

aperezhortal committed:
remove unsed shape argument from plot_geography

dnerini committed:
Speed up tests

dnerini committed:
Resample before flipping the array
